src/jit/lsraxarch.cpp

```diff
@@ -2530,81 +2588,74 @@ void LinearScan::BuildIndir(GenTreeIndir* indirTree)
 // Return Value:
 //    None.
 //
-void LinearScan::BuildMul(GenTree* tree)
+int LinearScan::BuildMul(GenTree* tree)
 {
-    TreeNodeInfo* info = currentNodeInfo;
-#if defined(_TARGET_X86_)
-    assert(tree->OperIs(GT_MUL, GT_MULHI, GT_MUL_LONG));
-#else
-    assert(tree->OperIs(GT_MUL, GT_MULHI));
-#endif
-    GenTree* op1   = tree->gtOp.gtOp1;
-    GenTree* op2   = tree->gtOp.gtOp2;
-    info->srcCount = appendBinaryLocationInfoToList(tree->AsOp());
-    assert(info->dstCount == 1);
+    assert(tree->OperIsMul());
+    GenTree* op1 = tree->gtOp.gtOp1;
+    GenTree* op2 = tree->gtOp.gtOp2;
 
-    // Case of float/double mul.
-    if (varTypeIsFloating(tree->TypeGet()))
-    {
-        return;
-    }
+    int       srcCount      = BuildRMWUses(tree->AsOp());
```

Comment:
Hmm, calling `BuildRMWUses` without calling `isRMWRegOper` that has special handling for `GT_MUL`? And FP MULs are never RMW.